sdk/src/androidTest/java/com/mongodb/stitch/TestUtils.kt

```diff
@@ -0,0 +1,55 @@
+package com.mongodb.stitch
+
+import com.google.android.gms.tasks.Task
+import com.mongodb.stitch.android.test.BuildConfig
+import java.util.concurrent.CountDownLatch
+
+/**
+ * Convenience method that allows suspension using a [CountDownLatch].
+ *
+ * @param[count] how many mutexes in this latch. defaults to 1
+ * @param[block] block of code to be executed. includes [CountDownLatch] context
+ *
+ * @return block result
+ */
+inline fun <R> latch(count: Int = 1, block: CountDownLatch.() -> R): R {
+    // instantiate a new latch with the count arg
+    val latch = CountDownLatch(count)
+    // call the block arg on the latch object
+    val r = latch.block()
+    // wait until the latch has been counted down
+    latch.await()
+    // return the result of the block
+    return r
+}
+
+/**
+ * Convenience method that wraps a [Task] in the [latch] method, awaiting the result.
+ *
+ * @param[task] pending task to wait for
+ *
+ * @return result of the task after it's been completed
+ */
+fun <T> await(task: Task<T>): T {
```

Comment:
https://kotlinlang.org/docs/reference/coroutines.html So after fiddling around with co-routines, I ran into a few issues, namely that they don't play well with the Tasks/Futures that we are currently using:

```kotlin
    inline fun <R> latch(count: Int = 1, block: CountDownLatch.() -> R): R {
        val latch = CountDownLatch(count)
        val r = latch.block()
        latch.await()
        return r
    }

    private suspend fun <T> fetch(task: Task<T>): T {
        latch { task.addOnCompleteListener { this.countDown() } }

        return task.result
    }

    @Test fun testAuth() = runBlocking {
        val deferredAuth = async(CommonPool) {
            fetch(stitchClient!!.logInWithProvider(AnonymousAuthProvider()))
        }

        val auth = deferredAuth.await()

        assert(auth.accessToken == FakeAccessToken)
    }
```

That was my experimentation. I ended up being able to accomplish all that in two lines without using coroutines:

```kotlin
@Test fun testAuth() {
        val auth = fetch(stitchClient!!.logInWithProvider(AnonymousAuthProvider()))

        assertThat(auth.accessToken == FakeAccessToken)
}
```

It looks like to take advantage of coroutines, we'd need to write our own Promise class in kotlin that uses the `suspended` keyword. It's possible we can implement our own `Task` class (`Task` is abstract after all), but that seems like overkill right now.

Could be a cool thought for a side project. Kotlin promises would be nice while the language is still in its nascent stages.